File: core/src/chrislo27/rhre/util/FileChooser.java
package chrislo27.rhre.util;

import chrislo27.rhre.Main;

import javax.swing.*;
import java.awt.*;

public class FileChooser extends JFileChooser {

	static {
		try {
			UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
		} catch (ClassNotFoundException | InstantiationException | IllegalAccessException |
				UnsupportedLookAndFeelException e) {
			e.printStackTrace();
			Main.logger.warn("Failed to set UIManager look-and-feel");
		}
	}

	@Override
	protected JDialog createDialog(Component parent) throws HeadlessException {
		JDialog dialog = super.createDialog(parent);
//		dialog.setLocationByPlatform(true);
		dialog.setAlwaysOnTop(true);
		return dialog;
	}

}

File: core/src/chrislo27/rhre/util/FileChooser.kt
package chrislo27.rhre.util

import java.awt.Component
import javax.swing.JDialog
import javax.swing.JFileChooser
import javax.swing.UIManager

open class FileChooser : JFileChooser() {

	override fun createDialog(parent: Component?): JDialog {
		val dialog = super.createDialog(parent)
//		dialog.setLocationByPlatform(true)
		dialog.isAlwaysOnTop = true
		return dialog
	}

	companion object {

		init {
			try {
				UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName())
			} catch (e: Exception) {
				e.printStackTrace()
				ionium.templates.Main.logger.warn("Failed to set UIManager look-and-feel")
			}

		}
	}

}
